src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random rand = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int number = rand.nextInt(5);
+        switch (number) {
+            case 1:
+                return new Circle(rand.nextInt(100),
+                        colorSupplier.getRandomColor());
+            case 2:
+                return new IsoscelesTrapezoid(rand.nextDouble(50),
+                        rand.nextDouble(50, 100),
+                        colorSupplier.getRandomColor());
+            case 3:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        rand.nextDouble(100),
+                        rand.nextDouble(150));
+            case 4:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        rand.nextInt(100),
+                        rand.nextInt(150));
```

Comment:
Avoid passing expressions directly into the constructor. Assign `colorSupplier.getRandomColor()`, `rand.nextInt(100)`, and `rand.nextInt(150)` to local variables before passing them to the `RightTriangle` constructor for better readability.